chain/src/chain.rs

```diff
@@ -550,11 +550,28 @@ impl Chain {
 	/// Meanwhile, the chain will not be able to accept new blocks. It should
 	/// therefore be called judiciously.
 	pub fn compact(&self) -> Result<(), Error> {
-		let mut sumtrees = self.txhashset.write().unwrap();
-		sumtrees.compact()?;
+		{
+			let mut txhashes = self.txhashset.write().unwrap();
+
+			// compact the txhashset
+			txhashes.compact()?;
+
+			// print out useful debug info after compaction
+			txhashset::extending(&mut txhashes, |extension| {
```

Comment:
Added `dump()` here for the output MMR after every call to the the `/compact` API endpoint so we can see what it did.